doc/design/codegen.md

```diff
@@ -0,0 +1,95 @@
+# SQLFlow Code Generator
+
+SQLFlow is a compiler that takes a SQL program and generates Argo workflow, which is a `.YAML` file,
+The Argo controller running on Kubernetes is the executor. This is a design doc on how to implement
```

Comment:
Actually, the term `executor` is rarely used in the compiler community. I suggest we find a more precise term for the concept. See 
https://en.wikipedia.org/wiki/Execution_(computing) for reference.